sources/zafira-services/src/main/java/com/qaprosoft/zafira/services/services/application/SettingsService.java

```diff
@@ -93,27 +106,53 @@ public void deleteSettingById(long id) throws ServiceException {
         return settingsMapper.getAllSettings();
     }
 
-    @Transactional(readOnly = true)
-    public List<Setting> getSettingsByIntegration(boolean isIntegrationTool) throws ServiceException {
-        return settingsMapper.getSettingsByIntegration(isIntegrationTool);
-    }
-
     public boolean isConnected(Tool tool) {
         return tool != null && !tool.equals(Tool.CRYPTO) && integrationService.getServiceByTool(tool).isEnabledAndConnected();
     }
 
-    @Transactional(readOnly = true)
-    public List<Tool> getTools() {
-        return settingsMapper.getTools().stream()
-                .filter(tool -> tool != null && !tool.equals(Tool.CRYPTO))
-                .collect(Collectors.toList());
-    }
-
     @Transactional(readOnly = true)
     public String getSettingValue(Setting.SettingType type) throws ServiceException {
         return getSettingByName(type.name()).getValue();
     }
 
+    @Transactional(rollbackFor = Exception.class)
+    public ConnectedToolType updateSettings(List<Setting> settings) throws ServiceException {
```

Comment:
Please refactor this method: introduce a couple of private methods with **meaningful** names. There are 2 times you throw an exception with relatively generic message. How about you use `Unable to update settings for multiple tools at once` (if that's correct according to my understanding of what's going on) for the first one and `Unable to update setting: setting does not belong to specified tool` for the 2nd (again if my understanding of code is correct)?